package/Dockerfile

```diff
@@ -1,3 +1,20 @@
+FROM --platform=$BUILDPLATFORM registry.suse.com/bci/golang:1.20 as build
```

Comment:
```suggestion
FROM --platform=$BUILDPLATFORM registry.suse.com/bci/golang:1.21 as build
```

Reply:
To align with https://github.com/rancher/rancher/pull/45674/files#diff-a5c0f08bc43f53a0451cad4a4a2e435ce55ad34183350dfae5e4b2c7dca6360bR14